vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -46,4 +53,46 @@ public ISlide getSlide(String slideId) {
     public void updateSlide(Slide slide) {
         slideRepo.save((Slide) slide);
     }
+
+    @Override
+    public void deleteSlideById(String slideId, String moduleId) throws SlideDoesNotExistException {
+
+        List<Sequence> sequences = sequenceRepo.findSequencesForModule(moduleId);
+        for(Sequence sequence : sequences) {
+
+            int sizeOfSlides = sequence.getSlides().size();
+            for(int i = 0; i< sizeOfSlides; i++) {
+                if(sequence.getSlides().get(i).getId().equals(slideId)) {
+                    sequence.getSlides().remove(i);
+                    sequenceRepo.save(sequence);
+                }
```

Comment:
so, i'm fairly certain this will have issues.

you are deleting elements from a list you're iterating over. So assume you remove the slide with index 0. Next iteration you take slide with index 1. At this point, the slide that previously had index 1, now has index 0 (because slide with index 0 was delete and all slides move one forward). Also, this will probably create an index out of bounds exception since if before deleting a slide the list had 3 slides, it now has 2 and a slide with i=2 will now not exist anymore. You can try it out easily:
```
        List<String> list = new ArrayList<>();
        list.add("aa");
        list.add("ab");
        list.add("bc");
        list.add("cd");
        
        int s = list.size();
        for(int i=0; i< s; i++) {
            System.out.println(i);
            if(list.get(i).startsWith("a")) {
                list.remove(i);
                System.out.println("deleted");
            }
        }
```
As I said before, you should use an iterator.

Reply:
Even if i am using iterator, I will still need to reference an index to retrieve the element.

Reply:
I am doing something like this.

int i = 0;
            Iterator<Sequence> sequenceIterator = sequences.iterator();
            while(sequenceIterator.hasNext()) {
                if(sequenceIterator.next().getSlides().get(i).getId().equals(slideId)){
                    sequenceIterator.next().getSlides().remove(i);
                    sequenceRepo.save(sequence);
                    sequenceIterator = sequences.iterator();
                    
                }
            }

Reply:
Also, I am not removing the list object. I am removing the slide object from the sequence list.
Hence, the loop should work as expected.